### PR TITLE
fix: request id validation

### PR DESCRIPTION
**Pull request by 0xJabberwock**
# 🤖 Linear

Closes GRT-3

### CODE REVIEW

File: solidity/interfaces/modules/resolution/IERC20ResolutionModule.sol [hunk 1]
```diff
@@ -53,6 +53,11 @@ interface IERC20ResolutionModule is IResolutionModule {
    */
   error ERC20ResolutionModule_OnlyDisputeModule();
```
Comment (0xJabberwock) [inline solidity/interfaces/modules/resolution/IERC20ResolutionModule.sol:54]: This custom error is not being used anywhere. 🙄

**Review by 0xJabberwock — COMMENTED** [review]
`TOB-WOND-4` explicitly mentions that the dispute's `requestId` validation issue could be found within `BondEscalationResolutionModule`, `BondEscalationModule`, `PrivateERC20ResolutionModule` and `ERC20ResolutionModule`. However, I was able to detect the lack of the same recommended check around many other modules. Should this be a concern?

Besides, other similar validations are also extendedly absent, such as comparing with the response's `requestId` or the dispute's `responseId`. Again, should this be a concern?

**Review by 0xShaito — DISMISSED** [review]

**Comment by 0xJabberwock** [comment]
> `TOB-WOND-4` explicitly mentions that the dispute's `requestId` validation issue could be found within `BondEscalationResolutionModule`, `BondEscalationModule`, `PrivateERC20ResolutionModule` and `ERC20ResolutionModule`. However, I was able to detect the lack of the same recommended check around many other modules. Should this be a concern?

It should not. This is due to every function lacking `requestId` checks being `onlyOracle`, contract where the appropriate validations are being performed.

> Besides, other similar validations are also extendedly absent, such as comparing with the response's `requestId` or the dispute's `responseId`. Again, should this be a concern?

It should not be it outside of the warned modules, for the same reason; within the warned modules, there was one case where a `responseId` validation was due.

**Review by 0xShaito — DISMISSED** [review]

File: solidity/test/unit/modules/resolution/BondEscalationResolutionModule.t.sol [hunk 2]
```diff
@@ -249,14 +249,16 @@ contract BondEscalationResolutionModule_Unit_PledgeForDispute is BaseTest {
 
     // block.timestamp < _startTime + _timeUntilDeadline
     _startTime = uint128(block.timestamp - timeUntilDeadline + 1);
-
-    (_requestId,, _disputeId) = _setResolutionModuleData(requestParameters);
   }
 
   function test_reverts(
```
Comment (ashitakah) [inline solidity/test/unit/modules/resolution/BondEscalationResolutionModule.t.sol:254]: Could you remane to specify the test function?

File: solidity/test/unit/modules/resolution/BondEscalationResolutionModule.t.sol [hunk 3]
```diff
@@ -526,14 +534,16 @@ contract BondEscalationResolutionModule_Unit_PledgeAgainstDispute is BaseTest {
...
```
Comment (ashitakah) [inline solidity/test/unit/modules/resolution/BondEscalationResolutionModule.t.sol:539]: same here

**Review by ashitakah — CHANGES_REQUESTED** [review]

**Review by 0xShaito — APPROVED** [review]

**Review by ashitakah — APPROVED** [review]

**Pull request merged**